### PR TITLE
Fixes mutation toxin runtime

### PR DESCRIPTION
**Pull request by nemvar**
## About The Pull Request

Mutation toxin used to delete itself before applying its effect. This fixes that by applying it on on_mob_add instead of on_mob_life
The actual mutate proc also used to be pretty dank.

### CODE REVIEW

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 1]
```diff
@@ -419,31 +419,29 @@
 	name = "Stable Mutation Toxin"
 	description = "A humanizing toxin."
 	color = "#5EFF3B" //RGB: 94, 255, 59
-	metabolization_rate = INFINITY //So it instantly removes all of itself
+	metabolization_rate = 0 //Doesn't metabolize. Gets destroyed on a timer.
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]: where does it get deleted?

Comment (nemvar) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm]: Good, point... could have sworn I did qdel(src) at the end of mutate but apparently didn't.

Comment (nemvar) [reply code/modules/reagents/chemistry/reagents/other_reagents.dm]: Done

**Comment by ExcessiveUseOfCobblestone** [comment]
so I haven't tried to break chemistry, but is there a hardstop to how far you can dilute a reagent to still get the add effects? I think I remember seeing one but it's been a while since I looked at the underbelly of reagents.

If there isn't this chem becomes practically infinite.

Prob an easier way to ensure this is a nonissue would be to require X units to be added at once prior to running the timer.

**Comment by ExcessiveUseOfCobblestone** [comment]
If so this is fine 4 mergies

**Comment by nemvar** [comment]
The chem can be dilluted as much as you want and it will still work. This is also an issue with some other things, for example unstable mutagen. Since that was the way it used to work, I didn't change it.

**Comment by ExcessiveUseOfCobblestone** [comment]
Fair enough, ill consider that OOS for this PR.

**Pull request merged**